internal/test/framework/windowsvm.go

```diff
@@ -190,49 +188,49 @@ func (w *windowsVM) RetrieveDirectories(remoteDir string, localDir string) error
 		// check if it is a directory, call itself again
 		if remoteFile.IsDir() {
 			if err = w.RetrieveDirectories(remotePath, localPath); err != nil {
-				log.Printf("error while retrieving directiories from Windows VM %s : %v", remotePath, err)
+				log.Printf("error while retrieving %s directiory from Windows : %v", remotePath, err)
 			}
 		} else {
-			// best effort to retrieve files
-			w.retrieveFiles(sftp, remotePath, localPath)
+			if err = w.copyFileFromRemote(sftp, remotePath, localPath); err != nil {
+				log.Printf("error while retrieving %s file from Windows : %v", remotePath, err)
+			}
 		}
 	}
 	return nil
 }
 
-// retrieveFiles retrieves list of files from remote directory to the local directory.
-// we're doing a best effort to collect every log possible. If a retrieval of file fails,
-// we would proceed with retrieval of other log files.
-func (w *windowsVM) retrieveFiles(sftp *sftp.Client, remotePath, localPath string) {
+// copyFileFromRemote copies a file from remote directory to the local directory.
+func (w *windowsVM) copyFileFromRemote(sftp *sftp.Client, remotePath, localPath string) error {
 
 	localFile, err := os.Create(localPath)
 	if err != nil {
-		log.Printf("error creating file locally: %v", err)
+		return fmt.Errorf("error creating file locally: %v", err)
 	}
 	// TODO: Check if there is some performance implication of multiple Open calls.
 	remoteFile, err := sftp.Open(remotePath)
 	if err != nil {
-		log.Printf("error while opening remote file on the Windows VM: %v", err)
+		return fmt.Errorf("error while opening remote file on the Windows VM: %v", err)
 	}
 	_, err = io.Copy(localFile, remoteFile)
 	if err != nil {
-		log.Printf("error retrieving file %v from Windows VM: %v", localPath, err)
+		return fmt.Errorf("error retrieving file %v from Windows VM: %v", localPath, err)
 	}
 	// flush memory
 	if err = localFile.Sync(); err != nil {
-		log.Printf("error flusing memory: %v", err)
+		return fmt.Errorf("error flusing memory: %v", err)
 	}
 	if err := remoteFile.Close(); err != nil {
-		log.Printf("error closing file on the remote host %s", localPath)
+		return fmt.Errorf("error closing file on the remote host %s", localPath)
 	}
 	if err := localFile.Close(); err != nil {
-		log.Printf("error closing file %s locally", localPath)
+		return fmt.Errorf("error closing file %s locally", localPath)
 	}
+	return nil
```

Comment:
Why are you returning the errors here? There was a reason why they were logs in the first place. For example if the `Copy`fails you still need to `Close` the files.

Reply:
sorry for the confusion, but I thought we want to bubble up the errors as much as possible. But I get the point that we need to close the files. I ll change them to logs.